### ISSUE TITLE
Further investigation on local network and discovery

### ISSUE THREAD

**Issue by tomoyukilabs** [httpslocal/proposals#7]
As pointed out in #1, device names in local network is not globally unique and it is difficult to determine how to trust such a device. Such a device often has its default name, it can be assigned automatically, or users can determine its name as they want.

On the other hand, we could face several difficulties when we would use such a local network device. For example:

- We must manually indicate device name as a form of a URL that begins with `http://` or `https://`.
- Current modern browser do not provide capability of service discovery to find local servers automatically. Note that the old versions of Safari for Mac used to have such a feature.
- Web applications do not have any methods to discover local servers due to security and privacy consideration. (see [W3C Working Group Note of Network Service Discovery](https://www.w3.org/TR/discovery-api/) for more information.)

In this issue, we would like to explore how we should deal with device names and DNS-SD/mDNS for local device utilization and enhancement without enlarging risks and threats, so that users would gain benefits of local device integration into web applications.

**Comment by tomoyukilabs** [comment]
One additional comment for clarification: Usually, default device names are defined by device manufacturers, and often like `VENDORNAME-MODELNAME-XXX.local` instead of a simple name like `printer.local`. Of course, such a name does not seem user-friendly.

**Comment by X-Ryl669** [comment]
If you can separate setup phase from using phase, then you can expect that a device will have its "unfriendly" name only during setup phase, and the user will give it a more useful name that'll be used in its using phase.

Ideally, the setup phase should use HTTPS, but the using phase must use HTTPS. If the proposal ensure the latter, I think it's good.

I don't think it's the goal of this proposal to make setup phase easier (setting up a new device is too *manufacturer-specific*), so let's focus on using phase being easy.

**Comment by tomoyukilabs** [comment]
> I don't think it's the goal of this proposal to make setup phase easier

@X-Ryl669 Correct. We should focus on the using phase. The comment above about the default name is just informative. Thanks for clarification.
